ddtrace/appsec/_iast/__init__.py

```diff
@@ -73,15 +70,12 @@ def ddtrace_iast_flask_patch():
 
 def enable_iast_propagation():
     """Add IAST AST patching in the ModuleWatchdog"""
-    if asbool(os.getenv(IAST.ENV, "false")):
-        from ddtrace.appsec._iast._utils import _is_python_version_supported
+    if _is_iast_enabled():
+        from ddtrace.appsec._iast._ast.ast_patching import _should_iast_patch
+        from ddtrace.appsec._iast._loader import _exec_iast_patched_module
 
-        if _is_python_version_supported():
-            from ddtrace.appsec._iast._ast.ast_patching import _should_iast_patch
-            from ddtrace.appsec._iast._loader import _exec_iast_patched_module
-
-            log.debug("IAST enabled")
-            ModuleWatchdog.register_pre_exec_module_hook(_should_iast_patch, _exec_iast_patched_module)
+        log.debug("IAST enabled")
```

Comment:
why do you remove `if _is_python_version_supported():`?

Reply:
Because `_is_iast_enabled()` already checks it